### PR TITLE
[Bugfix][SM120][MLA] Support FlashInfer packed sparse MLA decode

### PR DESCRIPTION
**Pull request by ChamHerry**
## Purpose

This PR fixes the released-wheel SM120 FlashInfer sparse MLA path used by GLM / Deepseek V3.2 style sparse MLA models with FP8 KV cache.

The current path can select the SM120 FlashInfer sparse MLA backend, but the decode API needs the packed `fp8_ds_mla` KV layout and `kv_scale_format` support. Separately, the sparse attention indexer only checked for the presence of DeepGEMM, so CUDA platforms where DeepGEMM paged MQA logits are not supported could still enter the wrong path or try to capture the torch fallback in CUDA graphs.

## Changes

- Update CUDA FlashInfer dependency to `flashinfer-python[cu13]==0.6.14` and document that CUDA 13 images should use `flashinfer-jit-cache` because `flashinfer-cubin 0.6.14` is not published.
- Canonicalize SM120 sparse MLA FP8 cache requests to `fp8_ds_mla`.
- Detect whether FlashInfer sparse MLA decode supports `kv_scale_format` and reject older FlashInfer builds explicitly.
- Pass packed uint8 KV cache, `seq_lens`, `bmm1_scale`, `bmm2_scale`, `sparse_mla_top_k`, and `kv_scale_format` to FlashInfer decode.
- Keep SM120 packed sparse MLA query inputs in BF16 instead of quantized query input.
- Add a DeepGEMM paged MQA capability helper and use it for indexer dispatch.
- Add an FP8 torch fallback for sparse attention indexer logits when DeepGEMM paged MQA is unavailable, while keeping FP4 indexer cache gated behind DeepGEMM support.
- Disable CUDA graph support for `DeepseekV32IndexerBackend` when the CUDA fallback path is active.
- Add targeted regression tests for the DeepGEMM capability helper, SM120 FlashInfer API gating/arguments, and sparse attention indexer fallback behavior.

## Duplicate-work check

I checked for related open PRs before opening this draft:

```bash
gh pr list --repo vllm-project/vllm --state open --search "FlashInfer SM120 sparse MLA" --limit 20
gh pr list --repo vllm-project/vllm --state open --search "fp8_ds_mla kv_scale_format" --limit 20
gh pr list --repo vllm-project/vllm --state open --search "DeepseekV32Indexer cudagraph SM120" --limit 20
gh pr list --repo vllm-project/vllm --state open --search "flashinfer 0.6.14 SM120" --limit 20
```

Nearest related PRs found:

- #46840 validates the SM120 sparse MLA top-k buffer path. It is related but does not update the packed `fp8_ds_mla` FlashInfer decode arguments, FlashInfer 0.6.14 dependency, or DeepGEMM paged MQA indexer fallback.
- #38476 adds a Triton sparse MLA backend/fallback. This PR keeps the FlashInfer SM120 sparse MLA backend and fixes its released FlashInfer 0.6.14 packed-KV path.
- #41834 is a broad DeepSeek V4 SM12x enablement branch. This PR is a narrower fix for the current main SM120 FlashInfer sparse MLA + Deepseek V3.2 indexer path.

I did not find an open PR that addresses this specific FlashInfer 0.6.14 packed sparse MLA decode and SM120 indexer fallback combination.

## Test Plan

Local formatting/static checks:

```bash
git diff --check
.venv/bin/pre-commit run --files \
  requirements/cuda.txt \
  vllm/utils/deep_gemm.py \
  vllm/utils/flashinfer.py \
  vllm/model_executor/layers/attention/mla_attention.py \
  vllm/model_executor/layers/sparse_attn_indexer.py \
  vllm/v1/attention/backends/mla/flashinfer_mla_sparse.py \
  vllm/v1/attention/backends/mla/flashinfer_mla_sparse_sm120.py \
  vllm/v1/attention/backends/mla/indexer.py \
  tests/utils/test_deep_gemm.py \
  tests/model_executor/layers/test_sparse_attn_indexer_deep_gemm.py \
  tests/v1/attention/test_flashinfer_sparse_mla_sm120_api.py
```

Remote SM120 Docker tests in an image rebuilt from `vllm/vllm-openai:latest-0701` with this branch overlaid and FlashInfer dependencies updated:

```bash
/usr/bin/python3 -m pytest \
  tests/utils/test_deep_gemm.py \
  tests/model_executor/layers/test_sparse_attn_indexer_deep_gemm.py \
  tests/v1/attention/test_flashinfer_sparse_mla_sm120_api.py \
  -v
```

Remote image dependency validation:

```text
flashinfer-python 0.6.14
flashinfer-jit-cache 0.6.14+cu130
flashinfer-cubin MISSING
cuda-tile 1.4.0
```

Remote GLM-5.2-NVFP4 service smoke test:

```bash
vllm serve GLM-5.2-NVFP4 \
  --tensor-parallel-size 8 \
  --max-model-len 131072 \
  --served-model-name GLM-5.2 GLM-5.2-STQ glm-5.2 glm-5.2-stq \
  --enable-prefix-caching \
  --tool-call-parser glm47 \
  --reasoning-parser glm45 \
  --enable-auto-tool-choice \
  --trust-remote-code
```

Validated endpoints:

```bash
curl http://127.0.0.1:4142/health
curl http://127.0.0.1:4142/v1/models
curl http://127.0.0.1:4142/v1/chat/completions ...
```

## Test Result

Passed:

```text
git diff --check: passed
pre-commit run --files ...: passed
pytest target suite: 12 passed, 16 warnings in 1.84s
remote service /health: ok
remote /v1/models: returned all served model names with max_model_len=131072
remote /v1/chat/completions: returned a successful response with content "你好！"
```

The same model with its default `max_model_len=1048576` did not fit in available KV cache on the test machine. Reducing `--max-model-len` to `131072` gave `GPU KV cache size: 319,424 tokens` and `Maximum concurrency for 131,072 tokens per request: 2.44x`, after which the service-level smoke test passed.

## AI Assistance Disclosure

AI assistance was used to investigate the runtime failures, implement this patch, run remote validation, and draft this PR description. This PR is opened as a draft; a human submitter should review every changed line and be prepared to defend the change end-to-end before marking it ready for review.

### CODE REVIEW

**Comment by liupengc** [comment]
+1
